srml/staking/src/lib.rs

```diff
@@ -1419,16 +1420,18 @@ impl<T: Trait> Module<T> {
 }
 
 impl<T: Trait> session::OnSessionEnding<T::AccountId> for Module<T> {
-	fn on_session_ending(_ending: SessionIndex, start_session: SessionIndex) -> Option<Vec<T::AccountId>> {
-		Self::new_session(start_session - 1).map(|(new, _old)| new)
+	fn on_session_ending(ending: SessionIndex, _: SessionIndex) -> Option<Vec<T::AccountId>> {
+		Self::new_session(ending + 1).map(|(new, _old)| new)
 	}
 }
 
-impl<T: Trait> OnSessionEnding<T::AccountId, Exposure<T::AccountId, BalanceOf<T>>> for Module<T> {
-	fn on_session_ending(_ending: SessionIndex, start_session: SessionIndex)
+impl<T: Trait> session::historical::OnSessionEnding<T::AccountId, Exposure<T::AccountId, BalanceOf<T>>>
+	for Module<T>
+{
+	fn on_session_ending(ending: SessionIndex, _: SessionIndex)
 		-> Option<(Vec<T::AccountId>, Vec<(T::AccountId, Exposure<T::AccountId, BalanceOf<T>>)>)>
 	{
-		Self::new_session(start_session - 1)
+		Self::new_session(ending + 1)
```

Comment:
This should either be `start_session` or `ending`, but not `ending + 1` or `start_session - 1`.

Reply:
Did you read the documentation?

Reply:
```suggestion
		Self::new_session(will_apply_at)
```